DEVELOPMENT.md

```diff
@@ -0,0 +1,43 @@
+# Development Guidelines
+
+This document describes the development guidelines when writing and maintaining EnMasse source code. 
+
+## General considerations
+
+As EnMasse is a polyglot project with multiple languages in the same repo, it uses `make` as a
+single entry point for invoking a full project build. Make will invoke the language-specific build 
+single entry point for invoking a full project build. Make will invoke the language-specific build
+system.
+
+### Introducing new dependencies
+
+If you need a function or class utility that requires adding a new dependency, first consider:
+
+* If the dependency has a reliable release schedule and track record for security updates.
+* If the dependency pulls in transitive dependencies that increase the footprint significantly
+* If you can copy the function or class (assuming the license allows it) into the EnMasse source tree. This should only be done for trivial amounts of code where the maintenance cost is expected to be low.
+* If the cost of implementing and maintaining the functionality yourself is worthwhile in the same manner as above.
+
+## Java
+
+Java source code is built using Maven.
+
+Follow the [Oracle Java Code Conventions](https://www.oracle.com/technetwork/java/codeconvtoc-136057.html).
```

Comment:
Taking a look at the link:

>  The information on this page is for Archive Purposes Only
>
>This page is not being actively maintained. Links within the documentation may not work and the information itself may no longer be valid. The last revision to this document was made on April 20, 1999

Can't we come up with something better?

Reply:
Did you have anything specific in mind?